### DISCUSSION TITLE
PRE-RFC: Add getters and setters for global variables

### DISCUSSION THREAD

**Discussion by sampersand** [ruby/rbs#1608] [Ideas]
This is quite niche, so it's probably not super important but...

Global variables in Ruby have "getters" and "setters". For user-defined global variables, the type of the global is the same when both getting and setting. However, quite a few builtin globals (`$.`, `$-F`, `$0`, etc.) can be assigned more permissive types, and convert them. (For example, `$.` can be assigned `int`s, but converts them internally.)

I propose "global variable setter and getter" syntax:
```rbs
$.: Integer
$.:= int

$0: String 
$0:= string

$-F: String | Regexp | nil
$-F:= string | Regexp | nil
```

If no setter is supplied, the global variable is assumed to have both the same getter and setter.

**Comment by ParadoxV5** [comment]
should be
```rbs
$.: Integer
$.=: int

$0: String 
$0=: string

$-F: String | Regexp | nil
$-F=: string | Regexp | nil
```
to match `def attribute=: …`

**Reply by sampersand** [reply]
That could work, but what about `$=`? `$==` feels a bit weird like it could be an actual variable

**Reply by ParadoxV5** [reply]
`$` isn’t a global variable is it… (;
`$==` … will this become a global variable itself…
